### PR TITLE
Defined USE_64BIT_TYPES for FPC 64-bit

### PR DESCRIPTION
**Pull request by denistotoliciu**
Pull request coordinated with the "Added IFDEFs to make JCL compatible with FPC 64-bit" pull request from the JCL repository.

### CODE REVIEW

File: jedi.inc [hunk 1]
```diff
@@ -1590,6 +1590,10 @@
     {$DEFINE SUPPORTS_OUTPARAMS}
     {$DEFINE SUPPORTS_WIDECHAR}
     {$DEFINE SUPPORTS_WIDESTRING}
+    {$IFDEF CPU64}
+		{$DEFINE USE_64BIT_TYPES}
+	{$ENDIF}
+	{$IFDEF HASINTF}
```
Comment (obones) [inline jedi.inc:1596]: The correct indentation is two spaces, not four, not tabs
And the last `IFDEF` will most likely render the file unusable as it is not balanced with a closing `ENDIF`

File: jedi.inc [hunk 2]
```diff
@@ -1590,6 +1590,10 @@
     {$DEFINE SUPPORTS_OUTPARAMS}
     {$DEFINE SUPPORTS_WIDECHAR}
     {$DEFINE SUPPORTS_WIDESTRING}
+    {$IFDEF CPU64}
+		{$DEFINE USE_64BIT_TYPES}
```
Comment (obones) [inline jedi.inc:1594]: What's the point of this define? It's not even used by your other pull request, and I don't see how a 64 bits CPU forces you to use 64bits types everywhere. This is not true for x86_64 and may not even be true for other CPU types.

Comment (pascaldragon) [reply jedi.inc:1594]: May I ask what the purpose of the USE_64BIT_TYPES define is? It's only used further down in COMPILER16_UP, but it's not documented inside the comments at the top. 🤔

File: jedi.inc (hunk 1 above)
Comment (pascaldragon) [reply jedi.inc:1596]: If I had to guess the check for HASINTF is a merge remnant from before my changes at the end of December where I moved the check for the HASINTF define into a check for FPC 2.0 and 2.1.

Comment (denistotoliciu) [reply jedi.inc:1596]: - `USE_64BIT_TYPES` is used only in `JclPeImage.pas` at the 6631 line to define the same function `InternalReadProcMem` with a `Size: SIZE_T` for 64-bit and a `Size: Integer` for 32-bit. Seems that this conditional has been defined solely for the purpose of defining the `InternalReadProcMem` function;

- Looking at how `SIZE_T` is defined for 64-bit as a `QWord` and for 32-bit as a `DWord`, the `USE_64BIT_TYPES` conditional could be left aside and removed and define the `InternalReadProcMem` function having the problematic parameter simply defined as `Size: SIZE_T`;

- `{$IFDEF HASINTF}` is an inattention error

File: jedi.inc (hunk 2 above)
Comment (denistotoliciu) [reply jedi.inc:1594]: - if you agree, I can completely remove the `USE_64BIT_TYPES` from `jedi.inc` and leave the header of the function mentioned above as `function InternalReadProcMem(ProcessHandle: THandle; Address: DWORD; Buffer: Pointer; Size: SIZE_T): Boolean;`, using the `Size: SIZE_T` parameter;

- haven't checked whether `USE_64BIT_TYPES` is used in other packages of the JEDI project;

- there are several pull requests that are on hold, but separated them like this at the advice given from people on the newsgroup. They were saying that there should be separate pull requests for each bug fixed, as well as for having FPC compile JCL for both 32-bit and 64-bit. This is the reason for which at the moment `USE_64BIT_TYPES` doesn't seem to be used by any other pull requests. Those pull requests would be on their way, but as mentioned above, this conditional could be removed if it is not used in other JEDI packages.

**Comment by denistotoliciu** [comment]
Will do another pull request that will with the appropriate modifications from the start.

**Pull request closed without merging**